### ISSUE TITLE
Adapt Commands apis for consistency with new World apis

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#1562]
From Future Work of #1525.

**Comment by alice-i-cecile** [comment]
@cart there's quite a few of us who are enthusiastic about this change, and I think the API consistency is very valuable. Can we add this to the 0.5 milestone?

**Comment by cart** [comment]
Yup I dig it!

**Comment by cart** [comment]
Theres only one mismatch between Commands and World:

```rust
world.spawn().insert_bundle(SomeBundle { .. });

commands.spawn(SomeBundle { .. });
```

I'm thinking that moving to `commands.spawn().insert_bundle(SomeBundle {..})` would largely just result in more boilerplate. Ex: we don't have a single example that needs to spawn an empty entity.

I think that consistency is a good idea, but maybe we should pursue alternatives, such as making `world.spawn()` take a Bundle again, then add something like `world.spawn_empty()` and `commands.spawn_empty()`.

However there is the matter of `spawn(SomeBundle)` not explicitly calling out that the parameter is a bundle (which confuses newbies). We could consider adding a new `commands.spawn_bundle(SomeBundle)` command, then `commands.spawn()` could just create a new empty entity.

**Comment by cart** [comment]
Currently investigating how disruptive moving to an api that completely mirrors the relevant World functions (including the "type state" pattern). I don't want to rush that in, but its also pretty surface level. I largely want to see how painful it is to adapt the examples.

**Comment by cart** [comment]
Closed by #1703
